common/src/java/com/zimbra/common/httpclient/HttpClientUtil.java

```diff
@@ -67,16 +68,40 @@ public static HttpResponse executeMethod(HttpClient client, HttpRequestBase meth
                 .setDefaultCookieStore(cookieStore)
                 .build ();
             return httpClient.execute(method);
-           
+
         }
         return client.execute(method);
     }
-    
+
+    public static HttpResponse executeMethod(HttpClientBuilder clientBuilder, HttpRequestBase method, BasicCookieStore state, HttpClientContext context) throws HttpException, IOException {
```

Comment:
Where is this used? `HttpClientBuilder` is not thread safe, curious where that param will be coming from.

Reply:
Currently it is not used anywhere. I added it for future use.